File: CS-Examples/23_Worksheets/GetPaperSize.vb
Imports System.IO
Imports System.Text
Imports Spire.Xls

Namespace GetPaperSize

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of the Workbook class
            Dim workbook As New Workbook()

            ' Load the workbook from the specified file path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\WorksheetSample2.xlsx")

            ' Create a StringBuilder object to store the result
            Dim stringBuilder As New StringBuilder()

            ' Iterate through each worksheet in the workbook
            For Each sheet As Worksheet In workbook.Worksheets

                ' Get the page width and height from the PageSetup object of the worksheet
                Dim width As Double = sheet.PageSetup.PageWidth
                Dim height As Double = sheet.PageSetup.PageHeight

                ' Append the worksheet name, width, and height to the StringBuilder
                stringBuilder.AppendLine(sheet.Name)
                stringBuilder.AppendLine("Width: " & width & vbTab & "Height: " & height)
                stringBuilder.AppendLine()
            Next sheet

            ' Specify the output file name
            Dim output As String = "GetPaperSize.txt"

            ' Write the content of the StringBuilder to the output file
            File.WriteAllText(output, stringBuilder.ToString())

            ' Release the resources used by the workbook
            workbook.Dispose()
            'Launch the file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace

File: CS-Examples/23_Worksheets/GetPaperSize.cs
using System;
using System.IO;
using System.Text;
using System.Windows.Forms;
using Spire.Xls;

namespace GetPaperSize
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            // Create a workbook
			Workbook workbook = new Workbook();

            // Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\WorksheetSample2.xlsx");

            StringBuilder sb = new StringBuilder();
            

            foreach (Worksheet sheet in workbook.Worksheets)
            {
                // Get page widths
                double width = sheet.PageSetup.PageWidth;

                // Get page height
                double height = sheet.PageSetup.PageHeight;
                sb.AppendLine(sheet.Name);
                sb.AppendLine("Width: " + width + "\tHeight: " + height);
                sb.AppendLine();
            }

            //Save to Text file
            string output = "GetPaperSize.txt";
            File.WriteAllText(output, sb.ToString());

            //Launch the file
            ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}
